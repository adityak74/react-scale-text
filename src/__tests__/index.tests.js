import React from 'react';
import { shallow, mount } from 'enzyme';
import ScaleText from '../index';


describe('ScaleText', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<div><ScaleText><p>Some text</p></ScaleText></div>);
    expect(wrapper).toMatchSnapshot();
  });

  // TODO: Find reliable way to unit test with dependencies on DOM
  // Element proeperties such as clientHeight/Width, scrollHeight/Width
  // getBoundingClientRect(), etc.  jsdom doesn't polyfill those, as it
  // doesn't have a full rendering engine.
  /*it('renders correctly, with minFontSize', () => {
    const pStyles = { width: '100px', height: '100px' };
    const wrapper = mount(
      <div className="parent" style={pStyles}>
        <ScaleText minFontSize={12}><p>Some text</p></ScaleText>
      </div>
    );
    console.log('offsetWidth: ', wrapper.find('.parent').offsetWidth);
    const fontSize = parseFloat(wrapper.find('p').getDOMNode().style.fontSize, 10);
    expect(fontSize).toBeGreaterThanOrEqual(12);
    wrapper.unmount();
  });

  it('renders correctly, with maxFontSize', () => {
    const pStyles = { width: '800px', height: '800px' };
    const wrapper = mount(
      <div style={pStyles}><ScaleText maxFontSize={20}><p>Some text</p></ScaleText></div>
    );
    const fontSize = parseFloat(wrapper.find('p').getDOMNode().style.fontSize, 10);
    expect(fontSize).toBeLessThanOrEqual(20);
    wrapper.unmount();
  });*/
});
