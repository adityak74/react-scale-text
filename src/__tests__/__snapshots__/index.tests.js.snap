// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleText renders correctly 1`] = `
<div>
  <ScaleText
    maxFontSize={Infinity}
    minFontSize={-Infinity}
  >
    <p>
      Some text
    </p>
  </ScaleText>
</div>
`;
